src/ngraph/runtime/gpu/gpu_primitive_emitter.cpp

```diff
@@ -70,3 +75,11 @@ void GPUPrimitiveEmitter::cache(const std::string& hash, const size_t& index)
 {
     m_primitive_map.insert({hash, index});
 }
+
+size_t GPUPrimitiveEmitter::get_primitive_index(std::unique_ptr<gpu::primitive>& f,
```

Comment:
get_primitive_index -> cache_primitive (more descriptive)

Reply:
Sorry to nit pick on this but this function is doing two things, it is cacheing the primitive, and it is adding it to hash lookup table. Would `register_primitive` be better? We already have a public member named `cache` -- this should probably also be made private now.